src/storage/storage_manager.cpp

```
@@ -117,10 +119,10 @@ SingleFileStorageManager::SingleFileStorageManager(AttachedDatabase &db, string
     : StorageManager(db, std::move(path), read_only) {
 }
 
-void SingleFileStorageManager::LoadDatabase() {
+void SingleFileStorageManager::LoadDatabase(const idx_t block_alloc_size) {
 
 	if (InMemory()) {
-		block_manager = make_uniq<InMemoryBlockManager>(BufferManager::GetBufferManager(db));
+		block_manager = make_uniq<InMemoryBlockManager>(BufferManager::GetBufferManager(db), DEFAULT_BLOCK_ALLOC_SIZE);
```

Maybe throw an exception if `block_alloc_size` is set for the in-memory case?

---

`ATTACH ':memory:' (BLOCK_SIZE 3)`